ursa/graph/utils.py

```diff
@@ -44,3 +46,50 @@ def _apply_append(collection, values):
             collection.update(val)
 
         return collection
+
+
+@ray.remote
+def write_vertex(vertex, graph_id, key):
+    dest_directory = graph_id + "/" + key + "/"
+
+    # @TODO(kunalgosar): Accessing _transaction_id here is not recommended
+    dest_file = dest_directory + str(vertex._transaction_id) + ".dat"
+    if not os.path.exists(dest_directory):
+        os.makedirs(dest_directory)
+
+    oids = [vertex.oid, vertex.local_keys]
+    foreign_keys = list(vertex.foreign_keys.keys())
+    oids.extend([vertex.foreign_keys[k] for k in foreign_keys])
+    oids = [pa.plasma.ObjectID(oid.id()) for oid in oids]
+
+    buffers = ray.worker.global_worker.plasma_client.get_buffers(oids)
+    data = {"node": buffers[0],
+            "local_keys": buffers[1],
+            "foreign_keys": foreign_keys,
+            "foreign_key_values": buffers[2:],
+            "transaction_id": vertex._transaction_id}
+
+    serialization_context = ray.worker.global_worker.serialization_context
+    serialized = pa.serialize(data, serialization_context).to_buffer()
+    with pa.OSFile(dest_file, 'wb') as f:
```

Comment:
`'wb'` -> `'ab'`. The order is going to be wrong, but you can fix that in the read.

Reply:
Currently each (Vertex, transaction_id) has its own file and so I write into a new file each time. Should I modify this so that each Vertex has its own file and I append in a new Vertex version each time? This might mean that accessing the nth version of a Vertex requires reading in a large file.

Reply:
You are correct, let's keep it how you have it.